Mage.Sets/src/mage/cards/a/AbaddonTheDespoiler.java

```diff
@@ -55,12 +50,16 @@ public AbaddonTheDespoiler(UUID ownerId, CardSetInfo setInfo) {
         this.addAbility(TrampleAbility.getInstance());
 
         // Mark of the Chaos Ascendant — During your turn, spells you cast from your hand with mana value X or less have cascade, where X is the total amount of life your opponents have lost this turn.
-        this.addAbility(new SimpleStaticAbility(new ConditionalContinuousEffect(
+        FilterSpell filter = baseFilter.copy();
+        Ability markOfTheChaosAscendant = new SimpleStaticAbility(new ConditionalContinuousEffect(
                 new GainAbilitySpellsEffect(new CascadeAbility(false), filter),
                 MyTurnCondition.instance, "during your turn, spells you cast from " +
                 "your hand with mana value X or less have cascade, where X is the " +
                 "total amount of life your opponents have lost this turn"
-        )));
+        ));
+        filter.add(new DynamicManaValuePredicate(ComparisonType.FEWER_THAN, new IntPlusDynamicValue(1, OpponentsLostLifeCount.instance),
```

Comment:
Stupid question 2: `markOfTheChaosAscendant` will save a reference to `filter` (line 54), and `filter` will save a reference to `markOfTheChaosAscendant` when you add the dynamic predicate. Won't this cause a circular reference and cause a memory leak?